source/dart/lib/astronomy.dart

```diff
@@ -88,3 +105,146 @@ class TerseVector {
+  /// not exactly equivalent; UTC and UT1 can disagree by up to plus or minus 0.9 seconds.
+  /// This approximation is sufficient for the accuracy requirements of Astronomy Engine.
+  ///
+  /// Universal Time Coordinate (UTC) is the international standard for legal and civil
+  /// timekeeping and replaces the older Greenwich Mean Time (GMT) standard.
+  /// UTC is kept in sync with unpredictable observed changes in the Earth's rotation
+  /// by occasionally adding leap seconds as needed.
+  ///
+  /// UT1 is an idealized time scale based on observed rotation of the Earth, which
+  /// gradually slows down in an unpredictable way over time, due to tidal drag by the Moon and Sun,
+  /// large scale weather events like hurricanes, and internal seismic and convection effects.
+  /// Conceptually, UT1 drifts from atomic time continuously and erratically, whereas UTC
+  /// is adjusted by a scheduled whole number of leap seconds as needed.
+  ///
+  /// The value in `ut` is appropriate for any calculation involving the Earth's rotation,
+  /// such as calculating rise/set times, culumination, and anything involving apparent
+  /// sidereal time.
+  ///
+  /// Before the era of atomic timekeeping, days based on the Earth's rotation
+  /// were often known as *mean solar days*.
+  final double ut;
+
+  /// Terrestrial Time days since noon on January 1, 2000.
+  ///
+  /// Terrestrial Time is an atomic time scale defined as a number of days since noon on January 1, 2000.
+  /// In this system, days are not based on Earth rotations, but instead by
+  /// the number of elapsed [SI seconds](https://physics.nist.gov/cuu/Units/second.html)
+  /// divided by 86400. Unlike `ut`, `tt` increases uniformly without adjustments
+  /// for changes in the Earth's rotation.
+  ///
+  /// The value in `tt` is used for calculations of movements not involving the Earth's rotation,
+  /// such as the orbits of planets around the Sun, or the Moon around the Earth.
+  ///
+  /// Historically, Terrestrial Time has also been known by the term *Ephemeris Time* (ET).
+  final double tt;
+
+  Time(this.ut, this.tt);
+
+  // For internal use only. Used to optimize Earth tilt calculations.
+  //double _psi = double.nan;
+
+  // For internal use only. Used to optimize Earth tilt calculations.
+  //double _eps = double.nan
+
+  // For internal use only. Lazy-caches sidereal time (Earth rotation).
+  //double st = double.nan
+
+  static Time fromUniversalTime(double ut) =>
+      Time(ut, ut /* turn this to terrestrialTime(ut) */);
+
+  /// Creates a `Time` object from a UTC year, month, day, hour, minute and second.
+  ///
+  /// @param year The UTC year value.
+  /// @param month The UTC month value 1..12.
+  /// @param day The UTC day of the month 1..31.
+  /// @param hour The UTC hour value 0..23.
+  /// @param minute The UTC minute value 0..59.
+  /// @param second The UTC second in the half-open range [0, 60).
```

Comment:
Dart documentation system doesn't have `@param` or `@return` or any alternative as far as I can say.

Reply:
This is one of the more time consuming things in Astronomy Engine development: generating useful documentation from the source code. In every language, I have had to spend significant time working on documentation generators. I try to use existing solutions as much as possible, but in the case of Python, I had to completely implement my own solution: [pydown.py](https://github.com/cosinekitty/astronomy/blob/master/generate/pydown/pydown.py). I may have to end up doing the same thing for Dart, if there is no markdown generator that understands parameters, return types, etc.

I keep thinking if someone wanted to make a popular GitHub project, it would be in the area of generating GitHub markdown for many different programming languages.